src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader_search.cs

```diff
@@ -84,6 +86,102 @@ private bool TryReadToSlow(out ReadOnlySpan<T> span, T delimiter, T delimiterEsc
             return true;
         }
 
+        private bool TryReadToSlow(out ReadOnlySequence<T> sequence, T delimiter, T delimiterEscape, int index, bool advancePastDelimiter)
+        {
+            BufferReader<T> copy = this;
+
+            ReadOnlySpan<T> remaining = UnreadSpan;
+            bool priorEscape = false;
+
+            do
+            {
+                if (index >= 0)
+                {
+                    if (index == 0 && priorEscape)
+                    {
+                        // We were in the escaped state, so skip this delimiter
+                        priorEscape = false;
+                        Advance(index + 1);
+                        goto Continue;
```

Comment:
Do we need to update `remaining` before continuing to the next iteration of the loop?

The same question would apply to the existing TryReadTo method:
https://github.com/dotnet/corefxlab/blob/b89fc2d9b404131b36ef6eb41320d638d4e4458c/src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader_search.cs#L161

If yes, we are missing a test case.

Reply:
Yes, we do, this is where you should actually write the test to try and break this. ;)  At the end of TryReadTo_SkipDelimiter, add the following:

``` C#
            // Delimiter starts second segment.
            bytes = BufferFactory.CreateUtf8("^", "|a|b");
            reader = new BufferReader<byte>(bytes);
            Assert.True(reader.TryReadTo(out span, (byte)'|', (byte)'^', advancePastDelimiter: true));
            Assert.Equal(Encoding.UTF8.GetBytes("^|a"), span.ToArray());
            Assert.True(reader.IsNext((byte)'b'));
            Assert.Equal(4, reader.Consumed);
```